src/main/java/com/ocelotslovebirds/birdhaus/datagen/RecipesGen.java

```diff
@@ -51,6 +52,10 @@ private static void birdhouseRecipe(
 
     @Override
     protected void buildCraftingRecipes(Consumer<FinishedRecipe> consumer) {
-        birdhouseRecipe(consumer, Registration.BIRDHOUSE_BLOCK.get(), BlockTags.LOGS, BlockTags.PLANKS);
+        birdhouseRecipe(consumer, Registration.BIRDHOUSE_BLOCK.get(), ItemTags.LOGS, ItemTags.PLANKS);
+    }
+
+    protected static InventoryChangeTrigger.TriggerInstance has(TagKey<Item> tagItem) {
+        return inventoryTrigger(ItemPredicate.Builder.item().of(tagItem).build());
     }
 }
```

Comment:
Ok cool this works! I was going suggest the usage of reflection but this works too :)

Reply:
if I understand the diff between private and protected correctly, wouldn't this method be better off as a private one, since we don't expect any other class to invoke it

Reply:
It would have to be protected since it is overriding the RecipeProvider's protected 'has' function and you can't have an override in Java that changes access level. However I think this brings up a really good point - there isn't really a need to override the has function just to re-implement tag possession. I changed the function name to possesses instead of 'has' and made it private, that way it is more clear that the function is only in the RecipesGen class and not quietly overriding the parent's function.